File: src/Domain/Entities/Event.cs
using Domain.Common;

namespace Domain.Entities;

public class Event : IEntity
{
    public int Id { get; set; }
    
    public string Name { get; set; }
    public string PageName { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
}

File: src/WebApi/ClientApp/src/services/models/Event.ts
export interface IEvent {
    id: number;

    name: string;
    pageName: string;
    startDate: Date;
    endDate: Date;
}

export default class Event implements IEvent {
  constructor(
    event: IEvent = {
      id: 0, 
      name: "",
      pageName: "",
      startDate: new Date(),
      endDate: new Date(),
    },
  ) {
    this.id = event.id;
    
    this.name = event.name;
    this.pageName = event.pageName;
    this.startDate = new Date(event.startDate);
    this.endDate = new Date(event.endDate);
  }

  id: number;

  name: string;
  pageName: string;
  startDate: Date;
  endDate: Date;
}
